petab/core.py

```diff
@@ -321,25 +438,52 @@ def to_float_if_float(x):
 
 
 def get_num_placeholders(formula_string, observable_id, override_type):
-    """Get number of unique placeholder variables in noise or observable definition"""
-
+    """
+    Get number of unique placeholder variables in noise or observable
+    definition.
+    """
+    
     pattern = re.compile(re.escape(override_type) + r'Parameter\d+_' + re.escape(observable_id))
     placeholders = set()
     for free_sym in sp.sympify(formula_string).free_symbols:
         free_sym = str(free_sym)
         if pattern.match(free_sym):
             placeholders.add(free_sym)
+    
     return len(placeholders)
 
 
 def get_dynamic_parameters_from_sbml(sbml_model):
-    """Get list of non-constant parameters in sbml model"""
-
+    """
+    Get list of non-constant parameters in sbml model.
+    TODO: This is not what Y would expect. Remove?
```

Comment:
What do you mean?

Reply:
So typically there would be parameters in the SBML model that are constant and never crop up in the parameter df? Can we then safely assume that it will be *exactly* those parameters that have the attribute ``constant``?
I implemented also a 2nd function, ``get_dynamic_simulation_parameters`` (
https://github.com/ICB-DCM/PEtab/blob/feature_map/petab/core.py#L515). Then one of those should be obsolete, right? 
I currently no-where use this method anyhow, but directly extract the parameters from the parameter df, but for automatic generation of a parameter df, as you said, having the method would be good.

Reply:
> So typically there would be parameters in the SBML model that are constant and never crop up in the parameter df?

Yes, some literature-based parameters for example.

> Can we then safely assume that it will be exactly those parameters that have the attribute constant?

Would also apply to "boundary conditions", would have to check for SBML details.

Reply:
Regarding the last part: Would keep both for now and clean up a bit later

Reply:
okay. since they aren't used anyway, shouldn't be a problem.